assets/shaders/terrain.wgsl

```diff
@@ -27,11 +28,23 @@ struct KdTree {
     count: u32,
 };
 
+struct Square {
```

Comment:
This should be a rectangle which may be rotated (not axis-aligned). This is because the game already support non-rectangular buildings and rotated (around vertical axis) buildings.

You can test the rotation by modifying a map, it stores angle for each building. New building construction will also support rotation in the future (#145).

We might want to represent it as a 3x3 matrix (with rotation + translation) and half_size 2D vector so we can quickly project `uv` point to the local coordinates of the rectangle (`projected = proj * vec3(uv, 1)`) and just test that `abs(projected) < half_size`. This way, we would need to pre-compute this for every building only when spawned (or more robustly, its `Transform` changed) which is way better than doing more calulations for every rendered pixel during every rendered frame.

Reply:
I made a commit of what I think that would look like, although it still ends up calculating a bunch of things per-pixel so I feel I might have missed a point somewhere